Add polling to reload configuration upon secrets changes.

manuel-guilbault opened the pull request:
Heavily inspired by the [AzureKeyVault extension](https://github.com/dotnet/extensions/tree/release/3.1/src/Configuration/Config.AzureKeyVault).

Addresses issue #23.

Kralizek commented:
Hi @manuel-guilbault,
Thank you for your contribution!

I gave it a quick look and it looks ok. I'll check it out properly at the end of my work day!

manuel-guilbault commented:
It looks like an `async` test I added was hanging when running using `dotnet test`, but runs successfully when running with VS test runner. I changed the test so it's not `async` anymore, and it works with `dotnet test` now.
